template.json

```
@@ -15,6 +21,9 @@
       "rules": {
         "prettier/prettier": "error"
       }
+    },
+    "lint-staged": {
+      "src/**/*.{js,jsx,ts,tsx,json}": ["npm run lint"]
```

I think you're overriding the staged files selection here by adding `./src`. `lint-staged` will pass the absolute path of every staged file that matches your glob.

```suggestion
      "src/**/*.{js,jsx,ts,tsx,json}": "eslint"
```